tests/integration/test_envelope.py

```diff
@@ -579,3 +579,51 @@ def test_sample_rates_metrics(mini_sentry, relay_with_processing, events_consume
 
     event, _ = events_consumer.get_event()
     assert event["_metrics"]["sample_rates"] == sample_rates
+
+
+# Checks that we buffer envelopes until we have a global config in processing mode.
+def test_buffer_envelopes_without_global_config(
+    mini_sentry, relay_with_processing, events_consumer
+):
+    include_global = False
+    original_endpoint = mini_sentry.app.view_functions["get_project_config"]
+
+    @mini_sentry.app.endpoint("get_project_config")
+    def get_project_config():
+        nonlocal include_global
+
+        res = original_endpoint().get_json()
+        if not include_global:
+            res.pop("global", None)
+        return res
+
+    mini_sentry.add_basic_project_config(42)
+    events_consumer = events_consumer()
+    options = {"cache": {"global_config_fetch_interval": 1}}
+    relay = relay_with_processing(options=options)
+
+    envelope = Envelope()
+    envelope.add_event({"message": "hello, world!"})
+    relay.send_envelope(42, envelope)
+
+    error_raised = False
+    try:
+        # Event is still in buffer because we have not provided a global config
+        events_consumer.get_event(timeout=1)
+    except AssertionError:
+        error_raised = True
+    assert error_raised
+
+    include_global = True
+    # Clear errors because we log error when we request global config yet we dont receive it.
+    assert len(mini_sentry.test_failures) > 0
+    for err in mini_sentry.test_failures:
+        assert (
+            str(err[1])
+            == "Relay sent us event: global config missing in upstream response"
+        )
+    mini_sentry.test_failures.clear()
+
+    # Global configs are fetched in 10 second intervals, so the event should have come
+    # through after a 10 sec timeout.
+    events_consumer.get_event(timeout=10)
```

Comment:
Please, assert that after the global config is fetched we get all sent envelopes here.

Reply:
do you mean to do it in a more explicit manner? because afaict, calling get_event does assert that we got the envelope, since it will panic if not present